src/frameworks/next/index.ts

```diff
@@ -210,10 +246,37 @@ export async function ɵcodegenPublicDirectory(sourceDir: string, destDir: strin
       }
     }
 
-    const prerenderManifestBuffer = await readFile(
-      join(sourceDir, distDir, "prerender-manifest.json")
-    );
+    const [prerenderManifestBuffer, routesManifestBuffer] = await Promise.all([
+      readFile(
+        join(
+          sourceDir,
+          distDir,
+          "prerender-manifest.json" // TODO: get this from next/constants
+        )
+      ),
+      readFile(
+        join(
+          sourceDir,
+          distDir,
+          "routes-manifest.json" // TODO: get this from next/constants
+        )
+      ),
+    ]);
+
     const prerenderManifest = JSON.parse(prerenderManifestBuffer.toString());
+    const routesManifest = JSON.parse(routesManifestBuffer.toString()) as Manifest;
+
+    const rewritesToUse = getNextjsRewritesToUse(routesManifest.rewrites);
+    const rewritesNotSupportedByFirebase = rewritesToUse?.filter(
+      (rewrite) => isRewriteSupportedByFirebase(rewrite) === false
+    );
+    const redirectsNotSupportedByFirebase = routesManifest.redirects?.filter(
+      (redirect) => isRedirectSupportedByFirebase(redirect) === false
+    );
+    const headersNotSupportedByFirebase = routesManifest.headers?.filter(
```

Comment:
If you guard with an empty array you can remove a bunch of the if-blocks later on.

```
const headersNotSupportedByFirebase = routesManifest.headers?.filter(
      (header) => isHeaderSupportedByFirebase(header) === false
    ) || [];


```